Classes/Util.php

```diff
@@ -274,22 +274,23 @@ public static function getConfigurationFromPageId(
 
         if ($initializeTsfe) {
             self::initializeTsfe($pageId, $language);
-            if (!empty($configurationToUse)) {
-                return self::buildTypoScriptConfigurationFromArray($configurationToUse, $pageId, $language, $path);
+            if (empty($configurationToUse)) {
```

Comment:
I would move this condition up one level since to avoid TSFE init when not required. Either that or I think a mistake may have snuck in here (see next comment below).